_docs/tasks/metrics-logs.md

```diff
@@ -9,210 +10,300 @@ type: markdown
+1. Create a new YAML file to hold configuration for the new metric and log
+   stream that Istio will generate and collect automatically.
+
+   Save the following as `new_telemetry.yaml`:
+   ```yaml
+   # Configuration for metric instances
+   apiVersion: "config.istio.io/v1alpha2"
+   kind: metric
+   metadata:
+     name: doublerequestcount
+     namespace: istio-config-default
+   spec:
+     value: "2" # count each request twice
+     dimensions:
+       source: source.service | "unknown"
+       destination: destination.service | "unknown"
+       message: '"twice the fun!"'
+     monitored_resource_type: '"UNSPECIFIED"'
+   ---
+   # Configuration for a Prometheus handler
+   apiVersion: "config.istio.io/v1alpha2"
+   kind: prometheus
+   metadata:
+     name: doublehandler
+     namespace: istio-config-default
+   spec:
+     metrics:
+     - name: double_request_count # Prometheus metric name
+       instance_name: doublerequestcount.metric.istio-config-default # Mixer instance name (fully-qualified)
+       kind: COUNTER
+       label_names:
+       - source
+       - destination
+       - message
+   ---
+   # Rule to send metric instances to a Prometheus handler
+   apiVersion: "config.istio.io/v1alpha2"
+   kind: rule
+   metadata:
+     name: doubleprom
+     namespace: istio-config-default
+   spec:
+     actions:
+     - handler: doublehandler.prometheus
+       instances:
+       - doublerequestcount.metric
+   ---
+   # Configuration for logentry instances
+   apiVersion: "config.istio.io/v1alpha2"
+   kind: logentry
+   metadata:
+     name: newlog
+     namespace: istio-config-default
+   spec:
+     severity: '"warning"'
+     timestamp: request.time
+     variables:
+       source: source.labels["app"] | source.service | "unknown"
+       user: source.user | "unknown"
+       destination: destination.labels["app"] | destination.service | "unknown"
+       responseCode: response.code | 0
+       responseSize: response.size | 0
+       latency: response.duration | "0ms"
+     monitored_resource_type: '"UNSPECIFIED"'
+   ---
+   # Configuration for a stdio handler
+   apiVersion: "config.istio.io/v1alpha2"
+   kind: stdio
+   metadata:
+     name: newhandler
+     namespace: istio-config-default
+   spec:
+    severity_levels:
+      warning: 1 # Params.Level.WARNING
+    outputAsJson: true
+   ---
+   # Rule to send logentry instances to a stdio handler
+   apiVersion: "config.istio.io/v1alpha2"
+   kind: rule
+   metadata:
+     name: newlogstdio
+     namespace: istio-config-default
+   spec:
+     match: "true" # match for all requests
+     actions:
+      - handler: newhandler.stdio
+        instances:
+        - newlog.logentry
+   ---
    ```
 
-   The expected output is:
+1. Push the new configuration.
 
    ```bash
-   Error: the server could not find the requested resource
+   kubectl -n istio-config-default apply -f new_telemetry.yaml
```

Comment:
istioctl

Reply:
istioctl doesn't support mixer rules, as far as i know.

Reply:
well, no, it turns out my version was just old enough not to support it.